### ISSUE TITLE
Add Shipping Profile and Upgrade Id to the Receipt

### ISSUE THREAD

**Issue by chris-rogala** [etsy/open-api#231]
**Is your feature request related to a problem? Please describe.**
The `shipping_method` was added but that is not informative at all. We have different shipping profiles that include USPS First Class or UPS Ground as the standard shipping. Currently, the the shipping_method is defined as:

> "shipping_method" - This will be "Standard Shipping" unless a shipping upgrade has been selected by the buyer.
(https://github.com/etsy/open-api/issues/211)

However, that is not helpful since the "Standard Shipping" is different between profiles:

![image](https://user-images.githubusercontent.com/44238629/134051664-6bb6aba5-7871-4c6a-8a14-b244f0e6602e.png)

**Describe the solution you'd like**
It would be a lot easier to manage the shipping if we know the shipping profile and upgrade for the transaction. I recommend adding this to the transaction or receipt like this:

```
{
    "shipping": {
        "shipping_profile_id": 123456,
        "upgrade_id": 7890123,
        "min_processing_days": 49,
        "max_processing_days": 63,
        "expected_ship_date": 1633741301,
        "actual_ship_date": null
    }
}
```

This was originally mentioned as a response to a comment here: https://github.com/etsy/open-api/issues/211

**Describe alternatives you've considered**
Pull the current item shipping profile and store that information statically. Since the data is changing constantly, there may be scenarios where the wrong shipping method is selected and this would result in a poor customer experience.

**Additional context**
N/A

**Comment by etsyachristensen** [comment]
@chris-rogala you are correct that the actual_ship_date and shipped_timestamp are the same. I spaced on that. We don't need the actual_ship_date field and I'll remove it. The shipping method *should* be the name on the shipping profile. However there are many shipping profiles that don't have names. I believe the was a time that shipping profiles could be created and didn't have names. If it comes out null, there's no name for the shipping profile. But you can still get the shipping profile from the shipping_profile_id field.

**Comment by chris-rogala** [comment]
@etsyachristensen I was working on pulling this information for orders we have shipping this week, but it appears the `shipping_profile_id` in the transactions does not correspond to a valid shipping profile. Running the below cURL will return a `shipping_profile_id` of 158500611722 for receipt 2264501910:

```
curl --location --request GET 'https://openapi.etsy.com/v3/application/shops/7974117/receipts/2264501910' \
--header 'x-api-key: {{api_key}}' \
--header 'Authorization: Bearer {{access_token}}' 
```
Result:
![2021-12-13_5-13-49](https://user-images.githubusercontent.com/44238629/145793678-b28a8fdf-ced1-4bbd-93db-9365b8ce2c84.jpg)

However, this shipping profile does not exist in the shipping profiles for our shop. See below:

![2021-12-13_5-16-40](https://user-images.githubusercontent.com/44238629/145794224-e41d8492-6838-4913-ad7f-7d6343da0695.jpg)

Based on the date this was purchased, the `shipping_profile_id` should be 159515426859. What is the fastest way this could be corrected? Thanks!

**Comment by chris-rogala** [comment]
@etsyachristensen  Doing more research and it appears the `shipping_profile_id` does equate to anything valid. I pulled two receipts of the same product purchased with the same purchase date, and the `shipping_profile_id` do not match, but they should match since the receipt is for the same listing and would have the same shipping profile assigned to it.

I have attached a screenshot for your reference. Also, the shopId is 7974117 and the receipts are 2267057608 and 2267134822. The` listing_id `purchased is 712584394 and the `shipping_profile_id` should be 159515426859.


![2021-12-13_5-26-06](https://user-images.githubusercontent.com/44238629/145796161-dba27257-cbe6-45a9-aaf9-fae320bc2853.jpg)
.

**Comment by chris-rogala** [comment]
@etsyachristensen @nateswart I hope you all are having a wonderful holiday season. I just wanted to bump this issue back to you all as this should be reopened. Hope to hear back from you all soon. Happy New Year!

**Comment by chris-rogala** [comment]
@etsyachristensen @nateswart I hope you all are having a wonderful holiday season. I just wanted to bump this issue back to you all as this should be reopened. Hope to hear back from you all soon. Happy New Year!
